scripts/pin_github_actions.py

```
@@ -0,0 +1,182 @@
+        if add_permissions_block(workflow_file):
+            updated_count += 1
+
+    print(f"\n✅ Updated {updated_count} workflow files")
+    print("\n⚠️  Important: Review all changes and test workflows before merging")
+    print("📝 Note: Some actions may need manual pinning if not in the known list")
+
+    return 0
+
+if __name__ == "__main__":
+    os.chdir("/home/omar/Documents/ruleIQ")
+    exit(main())
```

Pylint (3.3.8)

[refactor] Consider using 'sys.exit' instead

(R1722)